### PR TITLE
PLANET-6189 Implement the new Footer design

### PR DESCRIPTION
**Pull request by dantovbein**
:warning:
This PR is replaced with [this](https://github.com/greenpeace/planet4-master-theme/pull/1465)

Ref: https://jira.greenpeace.org/browse/PLANET-6189

---
**Design**
This is the new version of the Footer site based in these [versions](https://p4-designsystem.greenpeace.org/05f6e9516/p/16a899-footer).

**Affected versions:**
- Footer S, M, L and XL
- Minimal Footer S and L. Because we haven't any mockup for minimal's version, we reported it to the Designer and suggested a potential layout and it was approved by Magali. Here is the [thread of the conversation](https://greenpeace-gpi.slack.com/archives/G015K63081W/p1626186385006500)

**What is done.**
- Added a new Facebook and Linkedin rounded icons styled by adding this class name  `[social-name]-rounded` format. i.e. `facebook-rounded`.
- Improve the css using BEM for class names.

**Related PRs**
- [Minimal Footer design](https://github.com/greenpeace/planet4-master-theme/pull/1452)

:warning: **Attention!** 
- The acceptance test is failing because the buttons of the current Footer are upper cased and we won't need that format anymore.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [pandora](https://www-dev.greenpeace.org/test-pandora)

:watch: 2021.07.13 15:58:48

File: templates/footer.twig [hunk 1]
```diff
@@ -1,74 +1,80 @@
...
-						   data-ga-label="{{ social.title }}">
-							{{ name|svgicon }}
-							<span class="visually-hidden">{{ social.title }}</span>
-						</a>
-					</li>
-				{% endfor %}
-			{% endif %}
-		</ul>
+{% set follow_us = __( 'Follow us', 'planet4-master-theme' ) %}
+{% set total_social_icons = social_overrides is defined and social_overrides|length ? social_overrides|length : footer_social_menu|length %}
```
[the remark this answers is not in the document]
Comment (Inwerpsel) [reply templates/footer.twig:2]: There's quite a lot of template and logic involved for the social icons, perhaps better to give this its own template? Would also make for a cleaner diff here as this new variable could be inside that template instead.

File: templates/footer.twig [hunk 2]
```diff
@@ -1,74 +1,80 @@
...
-						   data-ga-action="Social Icons"
-						   data-ga-label="{{ social.title }}">
-							{{ name|svgicon }}
-							<span class="visually-hidden">{{ social.title }}</span>
-						</a>
-					</li>
-				{% endfor %}
-			{% endif %}
-		</ul>
+{% set follow_us = __( 'Follow us', 'planet4-master-theme' ) %}
```
Comment (Inwerpsel) [inline templates/footer.twig:1]: What is the reason for assigning this to a variable? The `__` function is used quite generally, I don't see a reason why this instance should be assigned.

File: assets/src/scss/layout/_footer.scss [hunk 3]
```diff
@@ -1,240 +1,239 @@
 .site-footer {
   --footer-- {
     font-family: $roboto;
-    background: $dark-blue;
+    background-color: $dark-blue;
```
Comment (Inwerpsel) [inline assets/src/scss/layout/_footer.scss:4]: These use `background` intentionally, as it supports the same values as `background-color`, but also some more like gradients and background images.

File: assets/src/scss/layout/_footer.scss [hunk 4]
```diff
@@ -1,240 +1,239 @@
...
-  }
 
   a --footer--links-- {
-    color: inherit;
+    color: $white;
 
-    &:hover {
-      color: $white;
-    }
+    // &:hover {
```
Comment (Inwerpsel) [inline assets/src/scss/layout/_footer.scss:17]: This is here intentionally: `--footer--links--` in the selector turns all properties into variables. So where in the default theme they have the same color, it allows a theme to have a different hover color.

Comment (Inwerpsel) [reply assets/src/scss/layout/_footer.scss:17]: However this is also a weird case: there's other color variables for different links in the footer. So probably it's better to remove this one and ensure each part can be set individually.

**Review by Inwerpsel — COMMENTED** [review]
Are all these changes to `_footer.scss` needed to consolidate the template? Not saying we shouldn't refactor that file, but I feel we would be changing too many things at the same time (pretty much every line of CSS) and it would be easy to miss things.

File: templates/footer.twig (hunk 2 above)
Comment (GP-Dan-Tovbein) [reply templates/footer.twig]: I agree

File: templates/footer.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (GP-Dan-Tovbein) [reply templates/footer.twig:1]: I completely agree with you

File: templates/base.twig [hunk 5]
```diff
@@ -41,11 +41,7 @@
 	{% endif %}
 
 	{% block footer %}
-		{% if custom_styles.nav_type == 'minimal' %}
-			{% include 'footer_min.twig' %}
-		{% else %}
-			{% include 'footer.twig' %}
-		{% endif %}
+		{% include 'footer.twig' with { 'nav_type' : custom_styles.nav_type } %}
```
Comment (Inwerpsel) [inline templates/base.twig:44]: We can do this refactor in a separate PR to make the design changes easier to evaluate.
